include/seqan3/alignment/pairwise/detail/policy_optimum_tracker.hpp

```diff
@@ -14,50 +14,112 @@
+        optimal_coordinate = (is_better_score) ? std::move(current_coordinate) : optimal_coordinate;
+    }
+};
+
 /*!\brief Implements the tracker to store the global optimum for a particular alignment computation.
  * \ingroup pairwise_alignment
  *
  * \tparam alignment_configuration_t The type of the alignment configuration; must be a type specialisation of
  *                                   seqan3::configuration.
- *
+ * \tparam binary_updater_t The type of a binary update operation to update the alignment optimum; must model
+ *                          std::semiregular and must be invocable with two tuples each containing the score and
+ *                          the coordinate.
  * \details
  *
- * Implements the interface to track the alignment optimum. It updates the currently stored optimum if the
- * optimal score of the given cell compares greater or equal to the stored optimum.
+ * Implements the interface to track the alignment optimum. It updates the currently stored optimum using the
+ * binary update operation. The binary update operation is stored inside of the class and can have a state.
  * Special methods are offered to track any cell (for example when computing the local alignment), the last cell of a
  * column or a row (for example when using free-end gaps), or the final cell of the entire matrix (for example in the
  * standard global alignment).
- * Depending on the configuration they might or might not evaluate the stored optimum score of that cell to compare
- * it with the currently stored optimum.
  * The optimum needs to be reset in between alignment computations in order to ensure that the correct result is
  * tracked.
  */
-template <typename alignment_configuration_t>
+template <typename alignment_configuration_t, std::semiregular binary_updater_t>
 //!\cond
-    requires is_type_specialisation_of_v<alignment_configuration_t, configuration>
+requires is_type_specialisation_of_v<alignment_configuration_t, configuration>
 //!\endcond
 class policy_optimum_tracker
 {
-private:
+protected:
     //!\brief The configuration traits type.
     using traits_type = alignment_configuration_traits<alignment_configuration_t>;
     //!\brief The configured score type.
     using score_type = typename traits_type::score_type;
+    //!\brief The matrix coordinate type that is used to locate a cell inside of the alignment matrix.
+    using matrix_coordinate_type = typename traits_type::matrix_coordinate_type;
 
+    /*!\name Data parameters
+     * \{
+     */
     //!\brief The tracked score of the global optimum.
     score_type optimal_score{};
     //!\brief The matrix coordinate of the tracked optimum.
-    matrix_coordinate optimal_coordinate{};
+    matrix_coordinate_type optimal_coordinate{};
+    //!\brief The function object to compare and exchange the optimum.
+    binary_updater_t compare_and_set_optimum{};
+    //!\}
+/*!\name Switch parameters
+     * \{
+     */
```

Comment:
```suggestion
    /*!\name Switch parameters
     * \{
     */
```

What are "Switch" parameters?